### DISCUSSION TITLE
Traversing edge until root (variable length)

### DISCUSSION THREAD

**Discussion by jjrodenburg** [apache/age#135] [Q&A] [answered]
I have the following graph:

user1 -[:parent]-> user2 -[:parent] -> user3

I'm trying to get the path from user1 to user3 using the parent edge. The number of users in the hierarchy is dynamic, but they will always have the parent edge if applicable.

The cypher code I'm trying to use gives me the error that "variable length relationships are not supported". I'm pretty new to cypher, so perhaps there's a different way of achieving this?

`SELECT * FROM cypher('user_structure', $$ MATCH p=(a:User {id: '76f62bee'})-[:parent*]->(r:User {tenant_id: '4f89978f'}) RETURN p $$) as (type agtype);`

If I use the non-variable length version, I can get 1 level of path. I'd like the entire path as output.
non variable:
`SELECT * FROM cypher('user_structure', $$ MATCH p=(a:User {id: '76f62bee'})-[:parent]->(r:User {id: '4f89978f'}) RETURN p $$) as (type agtype);`

Any ideas?

**Comment by jrgemignani** [comment, marked as answer]
Variable length relationships (edges) will start being available in a patch to the master branch, likely by the end of next week. Assigning it to a path variable will likely be in one shortly following that.

john

**Reply by jjrodenburg** [reply]
Thanks!
